engine/src/ast/function_expr.rs

```diff
@@ -520,9 +515,9 @@ impl GetType for FunctionCallExpr<'_> {
     }
 }
 
-impl<'i, 's> LexWith<'i, &FilterParser<'s>> for FunctionCallExpr<'s> {
-    fn lex_with(input: &'i str, parser: &FilterParser<'s>) -> LexResult<'i, Self> {
-        let (function, rest) = Function::lex_with(input, parser.scheme)?;
+impl<'i> LexWith<'i, &FilterParser<'_>> for FunctionCallExpr {
+    fn lex_with(input: &'i str, parser: &FilterParser<'_>) -> LexResult<'i, Self> {
+        let (function, rest) = FunctionRef::lex_with(input, parser.scheme)?;
```

Comment:
Similar here, should `lex_with` be implemented on `Function`, so we don't need to call `.to_owned()` on the `FunctionRef`?